wagtailmenus/templates/wagtailmenus/mainmenu_edit.html

```diff
@@ -1,20 +1,57 @@
-{% extends "modeladmin/create.html" %}
+{% extends "wagtailadmin/base.html" %}
```

Comment:
@dkirkham is there a Snippet template we can extend here instead that would already do some of this stuff? It would be nice to only override blocks where needed, if possible.

Reply:
Hi @ababic, my intent was to override only blocks, but I couldn't find a way. The issue was finding a place for the mainmenu select: the only feasible area was within wagtailadmin/shared/header.html and only a limited context is passed there. If you can see a way, let me know.

If we removed the mainmenu select (which would change editor/admin workflow a bit) then I think we could avoid changing any templates, but I considered that sort of change out of scope for this PR.

Reply:
Thanks for the explanation. Now that FieldPanel has the `read_only` option, I think we could get away with using that in the edit/create views to indicate the site, and perhaps only redirect straight to there in a single-site setup.